hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -205,6 +224,15 @@ public IndexedRecord next() {
         IndexedRecord record = this.reader.read(null, decoder);
         this.dis.skipBytes(recordLength);
         this.readRecords++;
+        if (whichImplementation) {
```

Comment:
What if `whichImplementation` is true but `castSchema` is not present?

Reply:
removed whichImplementation, but in the case castSchema is not present then we just return the record